conn_http_batch.go

```diff
@@ -0,0 +1,192 @@
+// Licensed to ClickHouse, Inc. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. ClickHouse, Inc. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package clickhouse
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/binary"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/column"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/proto"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+var splitHttpInsertRe = regexp.MustCompile(`(?i)INSERT INTO\s([\w.]+)(\s\((.*?)\))?`)
+var queryTableSchema = "SELECT name, type FROM system.columns WHERE table = $1 ORDER BY position"
```

Comment:
lets use DESCRIBE

Reply:
Fixed, but it produces more columns, that this query
https://github.com/ClickHouse/clickhouse-go/pull/622/files#diff-e23f0c1c92cf54758691c16198e28ad7511ff5c18c3542c343db1e8c4d9ccb11R62

Reply:
It was breaking change. Sometimes ttl_expression was nothing, EOF was return.

Moreover, `DESCRIBE` return 7 columns, but we need only 2.

And this query is analog of DESCRIBE, but return only needed columns

Reply:
it means people need access to system tables to use the driver though...

Reply:
I think use hack with `SETTINGS describe_include_subcolumns=1`. It always returns 0 or 1

Reply:
When `Decoder` reads a string of size 0 from `Reader` which points to the last byte. The `Reader` returns `EOF`.

Reply:
Added check [here](https://github.com/ClickHouse/clickhouse-go/pull/622/commits/04b2026b43bfe536e8792c1b95c2d67fb152b97b#diff-a813b8fd70923e6dfea0a06c69f24da5ac5ce47583cad16f0bc90c8081ae6e21R163)